corehq/apps/app_manager/suite_xml/post_process/endpoints.py

```diff
@@ -63,10 +63,20 @@ def _make_session_endpoint(self, module, form=None):
             stack=stack,
         )
 
-    def _get_argument_ids(self, frame_children):
+    def _get_argument_ids(self, frame_children, form):
+
+        def should_include(child):
+            if not isinstance(child, WorkflowDatumMeta):
+                return False
+            if child.requires_selection:
+                return True
+            if form:
+                return child.id in form.function_datum_endpoints or []
```

Comment:
This is going to return `[]` if the id isn't in `function_datum_endpoints`. I think you want:
```suggestion
                return child.id in (form.function_datum_endpoints or [])
```
if you are trying to protect against `function_datum_endpoints` being Falsey

Reply:
Won't jsonobject already default a `StringListProperty` to an empty list if the stored object doesn't have it defined?

Reply:
Thanks Farid. Fixed.

> Won't jsonobject already default a StringListProperty to an empty list if the stored object doesn't have it defined?

True. It gets populated via user input, so there could be a case where this might get set to None, which this will handle.